.github/workflows/markdown-toc.yml

```diff
@@ -0,0 +1,23 @@
+name: Markdown TOC
+
+on:
+  push:
+    paths:
+      - README.md
```

Comment:
Remove, table of content will be generated and committed on pull request trigger.

Reply:
But we also want to auto-update it when new commits are pushed directly to the main repo.

Reply:
`main` is protected. We will always go through pull requests I suggest.